apps/contact/forms.py

```diff
@@ -7,8 +7,21 @@ class ContactForm(forms.Form):
                     ("dotkom@online.ntnu.no", "Drifts- og utviklingskomiteen")
                     )
     contact_receiver = forms.ChoiceField(required=True, label="Hvem ønsker du å kontakte?", choices=mail_choices)
-    contact_name = forms.CharField(required=False, widget=forms.TextInput({"placeholder": "Navn"}))
-    contact_email = forms.EmailField(required=False, widget=forms.TextInput({"placeholder": "Epostadresse"}))
+    contact_checkbox = forms.BooleanField(required=False)
+    contact_name = forms.CharField(required=False, widget=forms.TextInput({"placeholder": "Navn",
+                                                                           "required": True}))
+    contact_email = forms.EmailField(required=False, widget=forms.TextInput({"placeholder": "Epostadresse",
+                                                                             "required": True}))
     content = forms.CharField(required=True, widget=forms.Textarea({"placeholder": "Din melding"}))
     captcha = ReCaptchaField(error_messages={'required': ('Du klarte ikke captchaen! Er du en bot?'),
                                              'invalid': ('Du klarte ikke captchaen! Er du en bot?')})
+
+    def clean(self):
+        name = self.cleaned_data.get('contact_name')
+        is_anon = self.cleaned_data.get('contact_checkbox')
+        email = self.cleaned_data.get('contact_email')
+
+        if not (name and email) and not is_anon:
+            raise forms.ValidationError(
+                "Du må skrive ditt navn og mail hvis du ikke skal være anonym!"
+            )
```

Comment:
I'd prefer if the error were attached to fields. If I'm not mistaken completely, this is accomplished by using `ValidationError({'field_name': 'Validation error text'})`. 

However, this resource shows the feature in a simpler way: https://docs.djangoproject.com/en/2.1/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other

If you use `self.add_error(field, text)` you can add the error to a specific field.

Could you look into that? 😄 

And conditionally check each field, obviously, so that you get one error per field instead of one error saying that one of the fields are missing.